components/compliance-service/config/config_mgmt.go

```diff
@@ -0,0 +1,90 @@
+package config
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/chef/automate/components/automate-deployment/pkg/toml"
+	base_config "github.com/chef/automate/lib/config"
+	log "github.com/sirupsen/logrus"
+)
+
+// ProjectUpdateConfig - the config for project updating
+type ProjectUpdateConfig struct {
+	State           string   `toml:"state"`
+	ProjectUpdateID string   `toml:"project_update_id"`
+	EsJobIDs        []string `toml:"es_job_ids"`
+}
+
+func defaultConfig() aggregateConfig {
+	return aggregateConfig{
+		ProjectUpdateConfig: ProjectUpdateConfig{},
+	}
+}
+
+// ConfigManager - configuration manager for the service
+type ConfigManager struct {
+	baseConfigManager *base_config.Manager
+}
+
+// Config - stores the configuration for the service
+type aggregateConfig struct {
+	ProjectUpdateConfig ProjectUpdateConfig `toml:"project_update_config"`
+}
+
+// NewConfigManager - create a new config. There should only be one config for the service.
+func NewConfigManager(configFile string) *ConfigManager {
```

Comment:
The other `ConfigManager` packages had a `NewManager` function for this purpose. Any reason this one needs to be different?

Reply:
There are a lot of calls in this PR to `NewManager`/`NewConfigManager` that are passed an empty string, which returns the default config, I'm guessing. Instead I'd suggest making that an explicit operation like `NewManagerWithDefaults()`/`NewManager(filename)` or `NewManager()`/`NewManagerFromFile(filename)`, etc.

Reply:
I named this `NewConfigManager` because the struct created is `ConfigManager`. This is not `Manager` because it was already taken in the types.go file. 

That makes sense to have the NewConfigManagerWithDefaults() function. I will add it.

Reply:
After looking at creating a `NewConfigManagerWithDefaults` function I don't think that would be a good idea, because a config file path needs to be passed in. All the uses of `NewConfigManager("")` are for tests, because we do not care if the file is saved.

Reply:
My biggest hesitance here is that due to an error or misconfiguration, we expect a string but don't get one and instead of an error we get some confusing default behavior.

Reply:
I agree with you now after talking about this in standup today. I made the changes in this commit https://github.com/chef/automate/pull/132/commits/0d8492f2ed178433a4031d3a631ce742c46f7dc0. Now if the file is not writable/readable an error is returned.